internal/pkg/service/objectstorage/cloud/azure/privateendpoint.go

```diff
@@ -0,0 +1,158 @@
+	s.logger.Info("Creating A record for private endpoint", "private-endpoint", *privateEndpoint.Name)
+
+	ips := make([]string, 0)
+
+	for _, dnsConfigs := range privateEndpoint.Properties.CustomDNSConfigs {
+		if dnsConfigs.IPAddresses == nil {
+			continue
+		}
+		for _, ip := range dnsConfigs.IPAddresses {
+			ips = append(ips, *ip)
+		}
+	}
+
+	aRecords := make([]*armprivatedns.ARecord, len(ips))
+	for i, ip := range ips {
+		aRecords[i] = &armprivatedns.ARecord{IPv4Address: &ip}
+	}
+
+	resp, err := s.recordSetsClient.CreateOrUpdate(
+		ctx,
+		s.cluster.GetResourceGroup(),
+		privateZoneID,
+		armprivatedns.RecordTypeA,
+		storageAccountName,
+		armprivatedns.RecordSet{
+			Properties: &armprivatedns.RecordSetProperties{
+				ARecords: aRecords,
+				TTL:      to.Ptr(int64(time.Hour.Seconds())),
+				Metadata: s.getBucketTags(bucket),
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &resp.RecordSet, nil
+}
+
+func (s AzureObjectStorageAdapter) upsertPrivateEndpoint(ctx context.Context, bucket *v1alpha1.Bucket, storageAccountName string) (*armnetwork.PrivateEndpoint, error) {
+	// Create or Update Private endpoint
+	pollersResp, err := s.privateEndpointsClient.BeginCreateOrUpdate(
+		ctx,
+		s.cluster.GetResourceGroup(),
+		bucket.Spec.Name,
+		armnetwork.PrivateEndpoint{
+			Location: to.Ptr(s.cluster.GetRegion()),
+			Properties: &armnetwork.PrivateEndpointProperties{
+				CustomNetworkInterfaceName: to.Ptr(fmt.Sprintf("%s-nodes-nic", bucket.Spec.Name)),
+				PrivateLinkServiceConnections: []*armnetwork.PrivateLinkServiceConnection{
+					{
+						Name: to.Ptr(bucket.Spec.Name),
+						Properties: &armnetwork.PrivateLinkServiceConnectionProperties{
+							PrivateLinkServiceID: to.Ptr(fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Storage/storageAccounts/%s", s.cluster.GetSubscriptionID(), s.cluster.GetResourceGroup(), storageAccountName)),
+							GroupIDs:             []*string{to.Ptr("blob")},
+						},
+					},
+				},
+				Subnet: &armnetwork.Subnet{
+					ID: to.Ptr(s.subnetID()),
+				},
+			},
+			Tags: s.getBucketTags(bucket),
+		},
+		nil,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+	resp, err := pollersResp.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp.PrivateEndpoint, nil
+}
+
+func (s AzureObjectStorageAdapter) upsertPrivateZone(ctx context.Context, bucket *v1alpha1.Bucket) (*armprivatedns.PrivateZone, error) {
+	pollersResp, err := s.privateZonesClient.BeginCreateOrUpdate(
+		ctx,
+		s.cluster.GetResourceGroup(),
+		privateZoneID,
+		armprivatedns.PrivateZone{
+			// Private Zone DNS is a global resource
+			Location: to.Ptr("Global"),
+			Tags:     s.getBucketTags(bucket),
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := pollersResp.PollUntilDone(ctx, nil)
```

Comment:
Does the `PollUntilDone` method create and/or update the the content of the `pollerResp` resource ?

Reply:
It calls the create endpoint then polls for a while every x seconds

Reply:
Ok thx